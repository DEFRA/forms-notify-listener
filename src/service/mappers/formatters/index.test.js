import { getFormatter } from '~/src/service/mappers/formatters/index.js'

jest.mock('nunjucks', () => {
  const environment = {
    addFilter: jest.fn(),
    addGlobal: jest.fn()
  }
  return {
    configure: jest.fn(() => environment)
  }
})
jest.mock('~/src/config/index.js', () => ({
  config: {
    get: jest.fn(() => {
      return 'http://designer'
    })
  }
}))

describe('Page controller helpers', () => {
  beforeAll(() => {
    jest.useFakeTimers().setSystemTime(new Date('2025-09-01T00:00:00Z'))
  })
  afterAll(() => {
    jest.useRealTimers()
  })

  it("should return an error if the audience doesn't exist", () => {
    expect(() => getFormatter('foobar', '1')).toThrow('Unknown audience')
  })

  it("should return an error if the version doesn't exist", () => {
    expect(() => getFormatter('human', '9999')).toThrow('Unknown version')
  })
})
