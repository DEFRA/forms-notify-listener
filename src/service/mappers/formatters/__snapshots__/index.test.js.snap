// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Page controller helpers should return a valid human readable v1 response 1`] = `
"^ For security reasons, the links in this email expire at 12:00am on Sunday 30 November 2025

Example notify form form received at 12:00am on 1 September 2025.

---

## What is your name?

Someone

---

## What is your address?

1 Anywhere Street
Anywhereville
Anywhereshire
AN1 2WH

---

## What is your date of birth?

1 January 2000

---

## What month is it?

August 2025

---

## What is the team member\\'s name?

[Download What is the team member\\\\'s name? \\(CSV\\)](http://designer/file-download/e3005cd2-8b1c-4dc4-b2ac-bd1ff73666a9)

---

## Who are your favourite LotR characters?

* Gandalf \\(Gandalf\\)
* Frodo \\(Frodo\\)

---

## Please add supporting evidence

Uploaded 1 file:

* [supporting\\_evidence\\.pdf](http://designer/file-download/ef4863e9-7e9e-40d0-8fea-cf34faf098cd)

---

[Download main form \\(CSV\\)](http://designer/file-download/818d567d-ee05-4a7a-8c49-d5c54fb09b16)
"
`;
